src/PaginateRoute.php

```diff
@@ -184,9 +183,30 @@ public function allUrls(LengthAwarePaginator $paginator, $full = false)
         }
 
         $urls = [];
-
-        for ($page = 1; $page <= $paginator->lastPage(); $page++) {
-            $urls[] = $this->pageUrl($page, $full);
+        $side = $paginator->onEachSide;
```

Comment:
There's a lot going on here. Could you extract this to it's own function. Also try to avoid using an `else` statement by using early returns.